python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -37,6 +39,10 @@ def _quantile_75(x):
 
 class GroupBy(Serializable, Reducible, Scannable):
 
+    # Pandas Index object supports groupby, but only trivially return a dict
+    # of grouped values.
```

Comment:
Is this comment intended to indicate that there won't be any subclass of GroupBy for Index objects? Out of curiosity, is there any way that isn't extremely painful to indicate to mypy that obj will be a DataFrame for DataFrameGroupBy and Series for SeriesGroupBy? Don't fight mypy too hard to make this work if you can't, but it's worth a quick shot.

Reply:
> Is this comment intended to indicate that there won't be any subclass of GroupBy for Index objects? 

Yes, I originally wrote this down as a self reminder. Should we remove it?

Reply:
Yeah I think this confuses more than it helps. It's basically saying "this is why X isn't present here", but without this comment I don't think most people would even look for X.